cpp/include/cudf/copying.hpp

```diff
@@ -939,5 +939,11 @@ std::unique_ptr<table> sample(
   int64_t const seed                  = 0,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+bool needs_sanitize(column_view const& input);
+
+std::unique_ptr<column> sanitize(
+  column_view const& input,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
```

Comment:
Seeing these names written down makes me feel we can do better with something more descriptive.

What comes to me at this moment is something like:
```
purge_nonempty_nulls
has_nonempty_nulls
may_have_nonempty_nulls
```

It looks like the current `needs_sanitize` does the expensive/exact check. In the original issue I described an approximate/fast check as well that we should add.

Reply:
Agreed. I'd prefer these explicit names myself.

Reply:
> In the original issue I described an approximate/fast check as well that we should add.

Re the approximate check: That would simply check if any of the columns in the hierarchy have null masks at all? Or should it check if `null_count() != 0`? The latter would involve launching kernels, but I suspect that's acceptable?

Reply:
> Or should it check if null_count() != 0? The latter would involve launching kernels, but I suspect that's acceptable?

That's more what I was thinking. It _may_ involve launching kernels. Recall that the `column::operator column_view` will compute the null count (if it doesn't already exist) during the first conversion and all later `column_view`s derived from that `column` will already have the null count.

Reply:
Kewl. ~`null_count()`~ `has_nulls()` it is.

Reply:
I've added a `may_have_nonempty_nulls()` function as well.